tests/test_pipeline_partition.py

```diff
@@ -51,5 +53,50 @@ def forward(self, x):
     assert ("stage2.linear.weight", 2) in tie_weights[0]
 
 
+def test_analyze_tie_ranks():
```

Comment:
I think you could just reuse the above test, as you also call `analyze_tie_weights` in this test.

Reply:
not really, the above test uses 3 stages, but we can only use 2 stages when there are only 2 gpus

Reply:
Huh I just realized that this test may not pass CI because we don't have DeepSpeed installed in the docker image...can you fake a topology? This also addresses the 2 GPU issue.